documentation/localization_howto.md

```diff
@@ -0,0 +1,165 @@
+added to the `streisand_languages` dict object in the `playbooks/common/vars/main.yml` file,
+where the 2 letter code is considered the key, and a list of key-value pairs with additional data
+as its value.
+
+For example:
+
+```
+streisand_languages:
+  en:
+    file_suffix: ""
+    language_name: "English"
+    tor_locale: "en-US"
+  fr:
+    file_suffix: "-fr"              # appends to the end of a file name
+    language_name: "Français"       # name of the language in the language itself
+    tor_locale: "fr"                # edge case for Tor          
+```
+
+Localizing gateway instructions
+-------------------------------
+
+The following roles contain user instructions that will require translation:
+
+ - `playbooks/roles/l2tp-ipsec/templates/`
+  1. `instructions.md.j2`
+
+ - `playbooks/roles/openconnect/templates`
+  1. `instructions.md.j2`
+  1. `mirror.md.j2`
+
+ - `playbooks/roles/openvpn/templates`
+  1. `instructions.md.j2`
+  1. `stunnel-insructions.md.j2`
+  1. `mirror.md.j2`
+
+ - `playbooks/roles/shadowsocks/templates`
+  1. `instructions.md.j2`
+  1. `mirror.md.j2`
+
+- `playbooks/roles/ssh-forward/templates`
+  1. `instructions.md.j2`
+  1. `mirror.md.j2`
+
+ - `playbooks/roles/streisand-gateway/templates`
+  1. `index.md.j2`
+  1. `instructions.md.j2`
+  1. `firewall.md.j2`
+
+ - `playbooks/roles/streisand-mirror/templates`
+  1. `mirror-index.md.j2`
+
+ - `playbooks/roles/stunnel/templates`
+  1. `mirror.md.j2`
+
+ - `playbooks/roles/tor/templates`
+  1. `instructions.md.j2`
+  1. `mirror.md.j2`
+
+ - `playbooks/roles/wireguard/templates`
+  1. `instructions.md.j2`
+
+Language selection in generated pages
+-------------------------------------
+
+By default, Streisand's common header files contains language selection 
+(generated using the `streisand_languages` dict), and for most pages will be done
+automatically. Certain edge cases do exist where a jinja2 code block is
+explicitly needed with the generated html filename.
+
+These files all reside with the `streisand-gateway` role:
+
+ - `firewall.md.j2`
+ - `index.md.j2`
+ - `instructions.md.j2`
+
+```
+- - -
+{% for key, value in streisand_languages.items() %}
+  [{{ value.language_name }}]({{ streisand_server_name }}-*generated-html-file-name*{{ value.file_suffix }}.html)&nbsp;
+{% endfor %}
+- - -
+```
+
+The HTML filename can be recovered by looking at the relevant ansible task that generates the
+template.
+
+End user considerations
+-----------------------
+
+While we aim to have as-close-to-the-original-source translations, linguistic differences in
+expressions should be taken into consideration when it makes.
+
+A non-technical example would be a the following
+expression in English: "the straw that broke the camel's back".
+A literal translation into French
+would give: "la paille qui a cassé le dos du chameau", however contextually
+a similar expression exists in French: "C'est la goutte d'eau qui fait déborder le vase"
```

Comment:
I did not know this!! Awesome! :fleur_de_lis: :fr:

Reply:
The More You Know&#8482;